core/src/main/scala/zio/config/ConfigSource.scala

```diff
@@ -4,15 +4,39 @@ import zio.{ IO, ZIO }
 import zio.system.System.Live.system
 import java.{ util => ju }
 
-final case class ConfigValue[A](value: ::[A])
-
+/**
+ * A config value is a list of A, meaning, multiple A's can exist for a key K.
+ * This can happen if the same key occurs in the config-source multiple times and there may or may not be
+ * a value of type A associated with it.
+ *
+ * Another possibility is, the value itself is a list for a single key K.
+ *
+ * Both these scenarios can be easily represented as ::[Option[A]].
+ */
+final case class ConfigValue[A](value: ::[Option[A]])
+
+/**
+ *
+ * ConfigSource represents the source to the config-description, which can be used for reading the values.
+ *
+ * @param getConfigValue : For a key, ConfigValue may or may not exist.
+ *                       Or in the process of fetching it, the computation can fail with some unknown error
+ *
+ * @param sourceDescription: Every ConfigSource is associated with a description
+ */
 final case class ConfigSource[K, V](
-  getConfigValue: Vector[K] => IO[ReadErrorsVector[K, V], ConfigValue[V]],
+  getConfigValue: Vector[K] => IO[ReadError.Unknown[Vector[K]], Option[ConfigValue[V]]],
```

Comment:
```suggestion
  getConfigValue: Vector[K] => Task[Option[ConfigValue[V]]],
```

In the code that calls `getConfigValue`, pack it into `ReadError.Unknown[Vector[K]]`.

Reply:
Consider:

```scala
IO[Option[Throwable], ConfigValue[V]]
```

But only if treating `None` as `Throwable` is common.

Reply:
Will do this